### ISSUE TITLE
PG::UndefinedFunction: ERROR: function with OID 1148257 does not exist

### ISSUE THREAD

**Issue by aguynamedben** [seamusabshere/upsert#52]
Some of my offline workers run into this error when using upsert:

```
PG::UndefinedFunction: ERROR: function with OID 1148257 does not exist
```

The job fails, and later upon retry, everything works. I just wanted to file in case anybody else was seeing this with upsert. Thanks for your work on this library.

Stack is
- Postgres 9.3.4
- upsert 2.1.0
- Ruby 2.1.3, Rails 3.2, Sidekiq 2.16.1, but I don't think that matters

**Comment by aguynamedben** [comment]
I'm upgrading pg to the most recent version (0.18.2). I don't think this bug is related to pg. I'll post back here if I see the error again after pg 0.18.2 is deployed.

**Comment by aguynamedben** [comment]
For what it's worth I still see this all the time.

![image](https://cloud.githubusercontent.com/assets/107841/9482906/708eebe4-4b4e-11e5-8c6a-aff8ba41180b.png)

Postgres 9.5 will be released in Q42015 and [will have upsert](https://wiki.postgresql.org/wiki/UPSERT) if anybody cares.

**Comment by pnomolos** [comment]
@aguynamedben Are you still using Upsert?  If so, the next version of the gem will have native support for PG 9.5+ `INSERT ... ON CONFLICT` syntax :)

**Comment by aguynamedben** [comment]
@pnomolos Yes, in one place. I believe in a few areas we just switched to doing a transaction to get rid of the errors, possibly at the cost of absolute consistency.

That's is *very* awesome I will check it out. Thanks!
